src/main/java/com/spotify/reaper/storage/MemoryStorage.java

```diff
@@ -496,4 +497,40 @@ public int hashCode() {
     }
   }
 
+  @Override
+  public boolean takeLeadOnSegment(long segmentId) {
+    return true;
+  }
+
+  @Override
+  public boolean renewLeadOnSegment(long segmentId) {
+    return true;
+  }
+
+  @Override
+  public void releaseLeadOnSegment(long segmentId) {
+  }
+
+  @Override
+  public void storeHostMetrics(HostMetrics hostMetrics) {
+  }
+
+  @Override
+  public Optional<HostMetrics> getHostMetrics(String hostName) {
+    return Optional.absent();
+  }
+
+  @Override
+  public StorageType getStorageType() {
+    return StorageType.MEMORY;
+  }
+
+  @Override
+  public int countRunningReapers() {
+    return 1;
+  }
+
+  @Override
+  public void saveHeartbeat() {
```

Comment:
my bad. i didn't see that saveHeartbeat was also called by the non-fault-tolerant storages.
 ( ie {{ReaperApplication:224}} )